templates/terraform/examples/healthcare_hl7_v2_store_parser_config.tf.erb

```diff
@@ -0,0 +1,94 @@
+resource "google_healthcare_hl7_v2_store" "default" {
+  provider = google-beta
+  name    = "<%= ctx[:vars]['hl7_v2_store_name'] %>"
+  dataset = google_healthcare_dataset.dataset.id
+
+  parser_config {
+    allow_null_header  = false
+    segment_terminator = "Jw=="
+    schema = jsonencode({
```

Comment:
just noticed this- the argument to jsonencode looks like json already to me. do you actually need the jsonencode here? my impression from https://www.terraform.io/docs/configuration/functions/jsonencode.html is that it maps from  HCL -> json

Reply:
Yes, jsonencode returns as a JSON encoded *string*. If we don't do this it will be interpreted as a map and Terraform will fail to parse it.

Unfortunately it doesn't seem like Terraform supports arbitrary object/map as an input, else I would just use that instead of requiring it to be a string.

Reply:
You can use a heredoc to make a multi-line string- check out https://www.terraform.io/docs/providers/google/r/bigquery_table.html for an example

Reply:
So one reason I prefer jsonencode is it allows you to write the schema as a native HCL format too.

e.g.
```
# map in JSON format
jsonencode({
  "foo": 1
})

# map in HCL format
jsonencode({
  foo = 1
})
```

are equivalent. If you think the heredoc way is better I can switch it.

I left this suggestion as well in https://github.com/terraform-providers/terraform-provider-google/issues/910#issuecomment-598474378

Reply:
But given this example uses JSON format it doesn't help to make Terraform waste time encoding it, I'll just switch to heredoc.